test/widgets/message_list_test.dart

```diff
@@ -608,6 +626,19 @@ void main() {
           narrow: narrow, messages: [message], unreadMsgs: unreadMsgs);
         check(isMarkAsReadButtonVisible(tester)).isTrue();
 
+        int unreadsNotifiedCount = 0;
+        final unreadsModel = store.unreads;
+        unreadsModel.addListener(() {
+          unreadsNotifiedCount++;
+        });
+        // Might as well test with oldUnreadsMissing: true.
+        //
+        // We didn't fill the model with 50k unreads, so this is questionably
+        // realistic… but the 50k cap isn't actually API-guaranteed, and this is
+        // plausibly realistic for a hypothetical server that decides based on
+        // message age rather than the 50k cap.
+        unreadsModel.oldUnreadsMissing = true;
```

Comment:
I think this "might as well" reasoning works so long as this wrinkle isn't adding so much to read (in code and comments) that it comes to take over the test case.

One simplification is that I think this test is just as effective without the `unreadsNotifiedCount` part, just checking that `oldUnreadsMissing` is still true. So:

```suggestion
        // Might as well test with oldUnreadsMissing: true.
        store.unreads.oldUnreadsMissing = true;
```